### PR TITLE
[TE] Add customized event in ThirdEye frontend

### PR DESCRIPTION
**Pull request by jihaozh**
Add a customized event pop-up in ThirdEye frontend.
screenshot from 2018-04-05 10-13-14
screenshot from 2018-04-05 12-14-17

### CODE REVIEW

**Comment by jihaozh** [comment]
Added the screenshot.

**Comment by apucher** [comment]
We'll need an input for "countryCode" dimension values s well.

And while I'm also guilty of non-CSS, there's a few CSS classes you could add anyways - just look at the related metrics modal :-)

File: thirdeye/thirdeye-frontend/app/pods/components/modals/create-event-modal/component.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ }}
+ * @exports create-event-modal
+ * @author jihzhang
+ */
+
+import Component from '@ember/component';
+import {
+  get, set, getProperties
+} from '@ember/object';
+import {inject as service} from '@ember/service';
```
Comment (aaronucsd) [inline thirdeye/thirdeye-frontend/app/pods/components/modals/create-event-modal/component.js]: spacing: 
```
import { inject as service } from '@ember/service';
```

File: thirdeye/thirdeye-frontend/app/pods/components/modals/create-event-modal/component.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+import fetch from 'fetch';
+
+import createEventApi from 'thirdeye-frontend/utils/api/create-event';
+
+export default Component.extend({
+
+  session: service(), /**
+   * Default selected mapping type
```
Comment (aaronucsd) [inline thirdeye/thirdeye-frontend/app/pods/components/modals/create-event-modal/component.js]: comment needs it's own line. Not sure which method it belongs to. Can we not add it before `actions`?

File: thirdeye/thirdeye-frontend/app/pods/components/modals/create-event-modal/component.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    onExit() {
+      this.set('isShowingModal', false);
+    },
+
+    onSave() {
+      const {
+        startTime, endTime, eventName, countryCode
+      } = getProperties(this, 'startTime', 'endTime', 'eventName', 'countryCode');
+
+      const startTimeSinceEpoch = new Date(startTime).getTime();
```
Comment (aaronucsd) [inline thirdeye/thirdeye-frontend/app/pods/components/modals/create-event-modal/component.js]: any of the inputs required? If so, we need to confirm and not allow the user to submit.

File: thirdeye/thirdeye-frontend/app/pods/components/modals/create-event-modal/component.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+   * Default selected mapping type
+   * @type {String}
+   */
+  actions: {
+    /**
+     * Handles the close event
+     * @return {undefined}
+     */
+    onExit() {
+      this.set('isShowingModal', false);
```
Comment (aaronucsd) [inline thirdeye/thirdeye-frontend/app/pods/components/modals/create-event-modal/component.js]: we might be able to use a mut for this. But just leave it for now.

File: thirdeye/thirdeye-frontend/app/pods/components/modals/create-event-modal/component.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  get, set, getProperties
+} from '@ember/object';
+import {inject as service} from '@ember/service';
+
+import fetch from 'fetch';
+
+import createEventApi from 'thirdeye-frontend/utils/api/create-event';
+
+export default Component.extend({
+
```
Comment (aaronucsd) [inline thirdeye/thirdeye-frontend/app/pods/components/modals/create-event-modal/component.js]: You might be missing this class to style the modal correctly. See `app/pods/components/modals/entity-mapping-modal/component.js`

```
  /**
   * Custom classes to be applied to the entity modal table
   */
  classes: {
    theadCell: "te-modal__table-header"
  },

```

File: thirdeye/thirdeye-frontend/app/pods/components/modals/create-event-modal/component.js (hunk 1 above)
Comment (jihaozh) [reply thirdeye/thirdeye-frontend/app/pods/components/modals/create-event-modal/component.js]: fixed

File: thirdeye/thirdeye-frontend/app/pods/components/modals/create-event-modal/component.js (hunk 5 above)
Comment (jihaozh) [reply thirdeye/thirdeye-frontend/app/pods/components/modals/create-event-modal/component.js]: fixed

File: thirdeye/thirdeye-frontend/app/pods/components/modals/create-event-modal/component.js (hunk 2 above)
Comment (jihaozh) [reply thirdeye/thirdeye-frontend/app/pods/components/modals/create-event-modal/component.js]: fixed

File: thirdeye/thirdeye-frontend/app/pods/components/modals/create-event-modal/component.js (hunk 3 above)
Comment (jihaozh) [reply thirdeye/thirdeye-frontend/app/pods/components/modals/create-event-modal/component.js]: currently not. leave it for now.

**Pull request merged**